posts/2018-06-07-shared-state-in-fp.md

```diff
@@ -0,0 +1,372 @@
+---
+layout: post
+title: Shared State in Functional Programming
+
+meta:
+  nav: blog
+  author: gvolpe
+  pygments: true
+
+tut:
+  scala: 2.12.4
+  binaryScala: "2.12"
+  dependencies:
+    - org.scala-lang:scala-library:2.12.4
+    - org.typelevel::cats-core:1.1.0
+
+---
+
+Newcomers to Functional Programming are often very confused about the proper way to share state without breaking purity and end up having a mix of pure and impure code that defeats the purpose of having pure FP code in the first place.
```

Comment:
No need to capitalize "functional programming" and consider parenthesizing (FP) next to it so your future use of FP is clear

Reply:
This might be a good article to cite https://queue.acm.org/detail.cfm?id=2611829